### PR TITLE
Add a vendor filter to analytics-vendors.amp.html for easier testing.

### PR DESCRIPTION
**Pull request by lannka**

### CODE REVIEW

File: build-system/app.js [hunk 1]
```diff
@@ -662,13 +662,21 @@ app.get(['/examples/*.html', '/test/manual/*.html'], function(req, res, next) {
...
     // Extract amp-ad for the given 'type' specified in URL query.
-    if (req.path.indexOf('/examples/ads.amp') == 0 && req.query.type) {
+    if (req.path.indexOf('/examples/ads.amp.html') == 0 && req.query.type) {
       var ads = file.match(new RegExp('<(amp-ad|amp-embed) [^>]*[\'"]'
           + req.query.type + '[\'"][^>]*>([\\s\\S]+?)<\/(amp-ad|amp-embed)>', 'gm'));
       file = file.replace(
           /<body>[\s\S]+<\/body>/m, '<body>' + ads.join('') + '</body>');
     }
 
+    // Extract amp-ad for the given 'type' specified in URL query.
```
Comment (avimehta) [inline build-system/app.js]: update comment.

File: extensions/amp-analytics/integrating-analytics.md [hunk 2]
```diff
@@ -18,16 +18,16 @@ Before you can add your analytics service to AMP HTML runtime, you may need to:
...
-4. Submit a Pull Request with this patch, referencing the Intent-To-Implement issue.
-5. Update your service's usage documentation and inform your customers.
+1. Develop a patch that implements the following:
+    1. A new block in [vendors.js](0.1/vendors.js) including any options above and beyond the default, such as:
+        1. "vars": {} for additional default variables.
+        1. "requests": {} for requests that your service will use.
+        1. "optout": if needed.  We currently don't have a great opt-out system, so please reach out to help us design one that works well for you.
+    1. An example in the [examples/analytics-vendors.amp.html](../../examples/analytics-vendors.amp.html)
+    1. A line for your analytics service in the 'type' attribute section of the [amp-analytics](amp-analytics.md) usage reference.
+1. Test the new example you put in [examples/analytics-vendors.amp.html](../../examples/analytics-vendors.amp.html) to ensure the hits from the example are working as expected. For example, the data needed is being collected and displayed in your analytics dashboard.
```
Comment (avimehta) [inline extensions/amp-analytics/integrating-analytics.md]: optional: say something about type parameter for the example file?

File: build-system/app.js [hunk 3]
```diff
@@ -662,13 +662,21 @@ app.get(['/examples/*.html', '/test/manual/*.html'], function(req, res, next) {
...
+    if (req.path.indexOf('/examples/analytics-vendors.amp.html') == 0 && req.query.type) {
+      var analytics = file.match(new RegExp('<amp-analytics [^>]*[\'"]'
```
Comment (avimehta) [inline build-system/app.js]: optional: move the creation of regex out to a function?

File: examples/analytics-vendors.amp.html [hunk 4]
```diff
@@ -17,13 +17,72 @@
...
   <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
   <script async src="https://cdn.ampproject.org/v0.js"></script>
 </head>
 <body>
 
+<div class="filterBar">
+  <form action="analytics-vendors.amp.html" method="get" target="_top">
+    <label for="filter">Vendor filter</label>
+    <select id="filter" name="type">
+      <option>[Select a vendor]</option>
```
Comment (avimehta) [inline examples/analytics-vendors.amp.html]: It is not clear exactly why one needs to select a vendor. Maybe put it somewhere? 

"Select a vendor to show config for that vendor." or something along these lines. I initially thought that selecting a type will allow you to run only that analytics type.

Comment (lannka) [inline examples/analytics-vendors.amp.html]: your initial thought is actually right

File: extensions/amp-analytics/integrating-analytics.md (hunk 2 above)
Comment (lannka) [inline extensions/amp-analytics/integrating-analytics.md]: the dropdown UI is intuitive, isn't it? :-)

File: build-system/app.js (hunk 3 above)
Comment (lannka) [inline build-system/app.js]: done

File: build-system/app.js (hunk 1 above)
Comment (lannka) [inline build-system/app.js]: done

**Comment by lannka** [comment]
@avimehta PTAL

**Pull request merged**